File: src/routes/fs.ts
import { Router } from 'express';
import fs from 'fs/promises';
import { existsSync, mkdirSync } from 'fs';
import path from 'path';

export const fsRouter = Router();

const AMMO_DIR = '/home/flintx/ammo';
const PROMPTS_BASE = '/home/flintx/peacock/prompts';

// --- AMMO ---
fsRouter.get('/ammo', async (req, res) => {
  try {
    if (!existsSync(AMMO_DIR)) return res.json([]);
    const files = await fs.readdir(AMMO_DIR);
    res.json(files.filter(f => f.endsWith('.md') || f.endsWith('.txt') || f.endsWith('.json')));
  } catch (e) { res.status(500).json({ error: 'Failed to read ammo dir' }); }
});

fsRouter.get('/ammo/:file', async (req, res) => {
  try {
    const content = await fs.readFile(path.join(AMMO_DIR, req.params.file), 'utf-8');
    res.json({ content });
  } catch (e) { res.status(500).json({ error: 'Failed to read file' }); }
});

// --- PROMPT ARSENAL (V22 Structure) ---
fsRouter.get('/prompts/:phase', async (req, res) => {
  const { phase } = req.params;
  const dir = path.join(PROMPTS_BASE, phase);
  try {
    if (!existsSync(dir)) return res.json([]);
    const files = await fs.readdir(dir);
    const prompts = await Promise.all(files.filter(f => f.endsWith('.md')).map(async f => ({
      id: f,
      name: f.replace('.md', ''),
      phase,
      content: await fs.readFile(path.join(dir, f), 'utf-8')
    })));
    res.json(prompts);
  } catch (e) { res.status(500).json({ error: `Failed to read prompts for ${phase}` }); }
});

fsRouter.post('/prompts/:phase', async (req, res) => {
  const { phase } = req.params;
  const { name, content } = req.body;
  const dir = path.join(PROMPTS_BASE, phase);
  try {
    if (!existsSync(dir)) mkdirSync(dir, { recursive: true });
    await fs.writeFile(path.join(dir, `${name}.md`), content);
    res.json({ status: 'SECURED' });
  } catch (e) { res.status(500).json({ error: 'Failed to secure prompt' }); }
});

fsRouter.delete('/prompts/:phase/:name', async (req, res) => {
  const { phase, name } = req.params;
  const filePath = path.join(PROMPTS_BASE, phase, `${name}.md`);
  try {
    if (existsSync(filePath)) await fs.unlink(filePath);
    res.json({ status: 'PURGED' });
  } catch (e) { res.status(500).json({ error: 'Failed to purge asset' }); }
});

File: app/routes/fs.py
import os
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import List
from pathlib import Path

router = APIRouter()

AMMO_DIR = "/home/flintx/ammo"
START_DIR = "/home/flintx/peacock/start"
PROMPTS_BASE = "/home/flintx/peacock/prompts"

class PromptAsset(BaseModel):
    id: str
    name: str
    phase: str
    content: str

class PromptSaveRequest(BaseModel):
    name: str
    content: str

@router.get("/start")
async def get_start_files():
    try:
        path = Path(START_DIR)
        if not path.exists():
            return []
        files = [f.name for f in path.iterdir() if not f.name.startswith('.')]
        return files
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to read start dir")

@router.get("/start/{file_name}")
async def get_start_file(file_name: str):
    try:
        file_path = Path(START_DIR) / file_name
        return {"content": file_path.read_text(encoding="utf-8")}
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to read start file")

@router.get("/ammo")
async def get_ammo():
    try:
        path = Path(AMMO_DIR)
        if not path.exists():
            return []
        files = [f.name for f in path.iterdir() if f.suffix in ['.md', '.txt', '.json']]
        return files
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to read ammo dir")

@router.get("/ammo/{file_name}")
async def get_ammo_file(file_name: str):
    try:
        file_path = Path(AMMO_DIR) / file_name
        return {"content": file_path.read_text(encoding="utf-8")}
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to read file")

@router.get("/prompts/{phase}", response_model=List[PromptAsset])
async def get_prompts(phase: str):
    dir_path = Path(PROMPTS_BASE) / phase
    try:
        if not dir_path.exists():
            return []
        prompts = []
        extensions = ["*.md", "*.txt"]
        for ext in extensions:
            for f in dir_path.glob(ext):
                prompts.append(PromptAsset(
                    id=f.name,
                    name=f.stem,
                    phase=phase,
                    content=f.read_text(encoding="utf-8")
                ))
        return prompts
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Failed to read prompts for {phase}")

@router.post("/prompts/{phase}")
async def save_prompt(phase: str, request: PromptSaveRequest):
    dir_path = Path(PROMPTS_BASE) / phase
    try:
        dir_path.mkdir(parents=True, exist_ok=True)
        file_path = dir_path / f"{request.name}.md"
        file_path.write_text(request.content, encoding="utf-8")
        return {"status": "SECURED"}
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to secure prompt")

@router.delete("/prompts/{phase}/{name}")
async def delete_prompt(phase: str, name: str):
    file_path = Path(PROMPTS_BASE) / phase / f"{name}.md"
    try:
        if file_path.exists():
            file_path.unlink()
        return {"status": "PURGED"}
    except Exception as e:
        raise HTTPException(status_code=500, detail="Failed to purge asset")